net.js

```diff
@@ -68,6 +68,37 @@ SSB.getThread = function(msgId, cb) {
   })
 }
 
+SSB.isConnected = false
+SSB.callbacksWaitingForConnection = []
+function runConnectedCallbacks() {
+  while(SSB.callbacksWaitingForConnection.length > 0) {
+    const cb = SSB.callbacksWaitingForConnection.shift()
+    cb(SSB)
+  }
+}
+
+SSB.connected = function(cb) {
+  // Add the callback to the list.
+  SSB.callbacksWaitingForConnection.push(cb)
+
+  if(SSB.isConnected) {
+    // Already connected.  Run all the callbacks.
+    runConnectedCallbacks()
+  } else {
+    // Register for the connect event so we can keep track of it.
+    SSB.net.on('rpc:connect', (rpc) => {
+      // Now we're connected.  Run all the callbacks.
+      SSB.isConnected = true
+      runConnectedCallbacks()
+
+      // Register an event handler for disconnects so we know to trigger waiting again.
+      rpc.on('closed', () => SSB.isConnected = false)
```

Comment:
minor thing here, but if you are connected to multiple peers this not really correct. I think ssb.conn should have a way to know if you are still connected?

Reply:
So you're saying this is incorrect in the use case of:

1. Callback is registered.  isConnected == false
2. Peer 1 connects.  isConnected == true.  Callback is run once
3. Peer 2 connects.  isConnected == true already
4. Peer 1 disconnects.  isConnected == false.  This is where it's incorrect because Peer 2 is still connected.
5. Callback is registered.  Since isConnected == false, it doesn't get run as waits until another peer connects.

If that's what you're referring to, that makes sense, and reference counting should work.  I just based that on how ui/connected.js works.

Reply:
yep :)